storage/dataflux/range_splitter.go

```diff
@@ -31,12 +34,308 @@ type listRange struct {
 	endRange   string
 }
 
+// minimalIntRange specifies start and end range in base-10 form, along with the
+// minimal string length for the split range strings.
+type minimalIntRange struct {
+	startInteger  *big.Int
+	endInteger    *big.Int
+	minimalLength int
+}
+
+// generateSplitsOpts specifies the parameters needed to generate the split
+// range strings.
+type generateSplitsOpts struct {
+	minimalIntRange *minimalIntRange
+	numSplits       int
+	startRange      string
+	endRange        string
+}
+
 // newRangeSplitter creates a new RangeSplitter with the given alphabets.
-func newRangeSplitter(alphabet string) *rangeSplitter {
-	return &rangeSplitter{}
+func newRangeSplitter(alphabet string) (*rangeSplitter, error) {
```

Comment:
Can you give a brief overview (here or elsewhere in this file) of the splitting strategy?

Reply:
Two function newRangeSplitter and splitRange are used here. Added an overview to it.

Reply:
I think we are still missing a general overview of how a set of objects should be split by this algorithm (or perhaps a walkthrough of an example).

Reply:
Added walkthrough of an example.